diff --git a/include/Makefile.in b/include/Makefile.in
--- a/include/Makefile.in
+++ b/include/Makefile.in
@@ -1,4 +1,5 @@
 SOURCES = \
+	distversion.h \
 	accctrl.h \
 	access.idl \
 	aclapi.h \
diff --git a/include/distversion.h b/include/distversion.h
new file mode 100644
--- /dev/null
+++ b/include/distversion.h
@@ -0,0 +1,4 @@
+/* Generated compatibility header for the OEM Wine source offer. */
+#ifndef __WINE_DISTVERSION_H
+#define __WINE_DISTVERSION_H
+#endif
